src/model/include/iDynTree/Model/Model.h

```diff
@@ -295,6 +295,27 @@ namespace iDynTree
                                    const std::string & newLinkName, Link & newLink);
 
         /**
+         * Insert a joint to the model, and add also a link. The other link in the existing joint
+         * that is not the unmovalbe link is the movable link.
+         * The new link replace the movable link as an attached link to the existing joint.
+         * Then a new joint is added between the new link and the movable link.
+         * The setAttachedLinks of the passed joint
+         * and the diplaced joint are edited appropriately, to ensure consistency in the model.
+         * @return the JointIndex of the added joint, or JOINT_INVALID_INDEX if
+         *         there was an error in adding the joint.
+         */
+        JointIndex insertJointAndLink(const std::string & existingJoint,
```

Comment:
I have to be honest, the name is not clear reading the description of what the method does.
Can you get a more explicative name? (for example, the fact that some links get replaced, etc..)
Do not fear long names.